addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1902,17 +1902,49 @@ var SnippetsMenu = Widget.extend({
             const $invisibleSnippets = globalSelector.all().find('.o_snippet_invisible').addBack('.o_snippet_invisible');
 
             $invisibleDOMPanelEl.toggleClass('d-none', !$invisibleSnippets.length);
-
-            const proms = _.map($invisibleSnippets, async el => {
-                const editor = await this._createSnippetEditor($(el));
+            // childrenPerSnippet -key: a snippet.
+            //                    -value: an array filled with its children
+            // snippets.
+            const childrenPerSnippet = new Map();
+            // Filter the $invisibleSnippets to only keep the non children
+            // snippets and create the map (childrenPerSnippet) of the snippets
+            // and their children.
+            const $rootInvisibleSnippets = $invisibleSnippets.filter((index, childSnippetEl) => {
+                for (const parentSnippetEl of $invisibleSnippets) {
+                    if (childSnippetEl !== parentSnippetEl
+                        && parentSnippetEl.contains(childSnippetEl)) {
+                        const childrenSnippets = childrenPerSnippet.get(parentSnippetEl) || [];
+                        childrenPerSnippet.set(parentSnippetEl, [...childrenSnippets, childSnippetEl]);
+                        return false;
+                    }
+                }
+                return true;
```

Comment:
I wonder if making a `filter` with a loop inside is not excessive (which makes O(n^2)). Couldn't you do something like that instead?
```js
for (const invisibleSnippet of $invisibleSnippets) {
    const ancestorInvisibleElement = invisibleSnippet.closest('.o_snippet_invisible');
    if (ancestorInvisibleElement) {
        childrenPerSnippet.get(ancestorInvisibleElement).push(invisibleSnippet);
    }
}
```